src/components/Map/CanadaMap.js

```diff
@@ -221,29 +262,68 @@ var CanadaMap = React.createClass({
     });
 
     map.addControl(new PrintControl());
-
-    map.addControl(new NcWMSColorbarControl(this.ncwmsLayer));
-    map.addControl(new NcWMSAutoscaleControl(this.ncwmsLayer, {
-      position: 'bottomright',
-    }));
+    
+    var autoscale;
+    //add colour legends and autoscale to the map
+    //if multiple colour legends are required (two layers), the
+    //autoscale button goes between them to indicate it affects 
+    //both layers
+    if(this.props.contourDataset && this.props.scalarDataset) {
+      map.addControl(new NcWMSColorbarControl(this.ncwmsContourLayer, {
+        position: 'bottomright'
+      }));
+      autoscale = new NcWMSAutoscaleControl(this.ncwmsScalarLayer, {
+        position: 'bottomright'
+      });
+      autoscale.addLayer(this.ncwmsContourLayer);
+      map.addControl(autoscale);
+      map.addControl(new NcWMSColorbarControl(this.ncwmsScalarLayer, {
+        position: 'bottomright'}));
+    }
+    else if(this.props.scalarDataset) {
+      map.addControl(new NcWMSAutoscaleControl(this.ncwmsScalarLayer, {
+        position: 'bottomright'
+      }));
+      map.addControl(new NcWMSColorbarControl(this.ncwmsScalarLayer, {
+        position: 'bottomright'
+      }));
+    }
+    else if(this.props.contourDataset) {
+      map.addControl(new NcWMSAutoscaleControl(this.ncwmsContourLayer, {
+        position: 'bottomright'
+      }));
+      map.addControl(new NcWMSColorbarControl(this.ncwmsContourLayer, {
+        position: 'bottomright'
+      }));
+    }    
```

Comment:
I _think_ the following might be easier to read and less repetitive:

```
    if(this.props.scalarDataset) {
      scalarAs = new NcWMSAutoscaleControl(this.ncwmsScalarLayer, {
        position: 'bottomright'
      });
      scalarCb = new NcWMSColorbarControl(this.ncwmsScalarLayer, {
        position: 'bottomright'
      });
    }
    
    if(this.props.contourDataset) {
      contourAs = new NcWMSAutoscaleControl(this.ncwmsContourLayer, {
        position: 'bottomright'
      });
      contourCb = new NcWMSColorbarControl(this.ncwmsContourLayer, {
        position: 'bottomright'
      });
    }
    
    if(this.props.contourDataset && this.props.scalarDataset) {
      map.addControl(scalarCb);
      scalarAs.addLayer(this.ncwmsContourLayer);
      map.addControl(scalarAs);
      map.addControl(contourCb);  
    }
    else if(this.props.contourDataset) {
      map.addControl(contourAs);
      map.addControl(contourCb);
    }
    else if(this.props.scalarDataset) {
        map.addControl(scalarAs);
        map.addControl(scalarCb);
    }
```

What do you think?